pyccel/parser/semantic.py

```diff
@@ -895,12 +901,25 @@ def _handle_kernel(self, expr, func, args, **settings):
                         symbol = expr,
                         severity='fatal')
             # TODO : type check the NUMBER OF BLOCKS 'numBlocks' and threads per block 'tpblock'
+            if not all(isinstance(param, (LiteralInteger, PythonTuple, PyccelSymbol))\
+                    for param in [expr.numBlocks, expr.tpblock]):
+                errors.report("Invalid parameter for Kernel Block number or Thread per Block",
+                        symbol = expr,
+                        severity='error')
+
             new_expr = KernelCall(func, args, expr.numBlocks, expr.tpblock, self._current_function)
-            if None in new_expr.args:
-                errors.report("Too few arguments passed in function call",
+
+            for a in new_expr.args:
+                if a is None:
```

Comment:
Yeah thats true ill fix that in a moment.

Reply:
May i ask how this works ? shouldn't it be checking if each arg has it corresponding value ?

Reply:
So how can we check if the right number of argument is passed to the function ?

Reply:
well we can use functionDef `arguments` attribute to check the number of passed arguments vs the number of arguments because from what i see we are just checking if nothing is passed to the function call.

Reply:
that should be already handled elsewhere